### PR TITLE
upstream sync

### PR DESCRIPTION
**Pull request by casperdcl**
- [ ] check notes
- [ ] fix conflicts

### CODE REVIEW

File: niftypad/kt.py [hunk 1]
```diff
@@ -69,23 +92,36 @@ def dt_fill_gaps(dt):
...
+        tac_inputf1linear = interpt1(mft, tac, dt)
+        tac_no_gaps = int2dt(tac_inputf1linear, dt_no_gaps)
+    if interp is 'cubic':
+        tac_inputf1cubic = interpt1cubic(mft, tac, dt)
+        tac_no_gaps = int2dt(tac_inputf1cubic, dt_no_gaps)
+    if interp is 'linear':
+        tac_inputf1linear = interpt1(mft, tac, dt)
+        tac_no_gaps = int2dt(tac_inputf1linear, dt_no_gaps)
+    if interp is 'zero':
+        tac_no_gaps = np.zeros_like(mft_no_gaps)
```
Comment (casperdcl) [inline niftypad/kt.py:117]:
```suggestion
    if interp in ['linear', 'cubic']:
        tac_inputf1 = {'linear': interpt1, 'cubic': interpt1cubic}[interp](mft, tac, dt)
        tac_no_gaps = int2dt(tac_inputf1, dt_no_gaps)
    elif interp == 'zero':
        tac_no_gaps = np.zeros_like(mft_no_gaps)
    else:
        raise ValueError(interp)
```

File: niftypad/models.py [hunk 2]
```diff
@@ -247,14 +288,21 @@ def logan_ref(tac, dt, inputf1, linear_phase_start, linear_phase_end, fig):
...
+def logan_ref_k2p(tac, dt, inputf1, k2p, w, linear_phase_start_l2, linear_phase_end_l2, fig):
+    if w is None:
+        w = np.ones_like(tac)
+    linear_phase_start = linear_phase_start_l2
+    linear_phase_end = linear_phase_end_l2
     if linear_phase_start is None:
         linear_phase_start = 0
     if linear_phase_end is None:
         linear_phase_end = np.amax(dt)
+    if linear_phase_end > np.amax(dt):
```
Comment (casperdcl) [inline niftypad/models.py:300]: what if linear_phase_end `< np.amax(dt)`? Or `< 0`? Or even `< linear_phase_start`?

Comment (JJiao) [reply niftypad/models.py:300]: -  linear_phase_end < np.amax(dt)
this is fine
- < 0
error
- < linear_phase_start
error

Comment (casperdcl) [reply niftypad/models.py:300]: ok so we should check for those and raise errors then, right?

**Pull request merged**